### DISCUSSION TITLE
pl/sql types

### DISCUSSION THREAD

**Discussion by wafflze** [sijms/go-ora#166] [Q&A]
Can I use this to set custom pl/sql types? I don't have direct access to the package and I don't want to use pl/sql directly and want to remove the following snippet:

```sql
declare

my_pl_sql_type PACKAGE.MY_PL_SQL_TYPE;

begin

my_pl_sql_type.FIRST_NAME := :1;
my_pl_sql_type.LAST_NAME := :2;

PACKAGE.insert_record(my_pl_sql_type, x_err_cd, _x_err_msg);

commit;

....
-- additional exception handling
....

end;

```

Ultimately, my questions are:
1. How to prepare local struct for a pl/sql record defined in package specifications.
2. How to pass local struct to execute pl/sql packaged procedure as one of the input parameters.

**Comment by sijms** [comment]
I think you talk about user defined types UDT

there is an 2 examples explaining this feature
1- UDT: show you how to user UDT with select
2- udt_pars: show you how to use UDT as output parameters
